### ISSUE TITLE
ERROR: UndefVarError: f not defined

### ISSUE THREAD

**Issue by jne555** [JuliaData/CSV.jl#27]
When doing a CSV.read I'm getting an undefined var error. It looks like the last commit changed the param name of "f" to "io", but didn't update the variables in the function "countlines".

`julia> CSV.read("../ndsparse_use_cases/relations/market_to_bu_relation.csv",
                                                  types = [String,String])
ERROR: UndefVarError: f not defined
 in countlines(::Base.AbstractIOBuffer{Array{UInt8,1}}, ::UInt8, ::UInt8) at /home/jnelson/.julia/v0.5/CSV/src/io.jl:79
 in #Source#3(::String, ::CSV.Options, ::Int64, ::Int64, ::Array{DataType,1}, ::Int64, ::Int64, ::Int64, ::Bool, ::Type{T}) at /home/jnelson/.julia/v0.5/CSV/src/Source.jl:76
 in (::Core.#kw#Type)(::Array{Any,1}, ::Type{CSV.Source}) at ./<missing>:0
 in #Source#2(::UInt8, ::UInt8, ::UInt8, ::String, ::Int64, ::Int64, ::Array{DataType,1}, ::Base.Dates.DateFormat, ::Int64, ::Int64, ::Int64, ::Bool, ::Type{T}, ::String) at /home/jnelson/.julia/v0.5/CSV/src/Source.jl:39
 in (::Core.#kw#Type)(::Array{Any,1}, ::Type{CSV.Source}, ::String) at ./<missing>:0
 in #read#4(::UInt8, ::UInt8, ::UInt8, ::String, ::Int64, ::Int64, ::Array{DataType,1}, ::Base.Dates.DateFormat, ::Int64, ::Int64, ::Int64, ::Bool, ::CSV.#read, ::String, ::Type{T}) at /home/jnelson/.julia/v0.5/CSV/src/Source.jl:274
 in (::CSV.#kw##read)(::Array{Any,1}, ::CSV.#read, ::String, ::Type{T}) at ./<missing>:0 (repeats 2 times)
 in eval(::Module, ::Any) at ./boot.jl:234
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46`

**Comment by quinnj** [comment]
Oops! Let me get that sorted out....

**Comment by ararslan** [comment]
Looks like this was fixed in https://github.com/JuliaData/CSV.jl/commit/dd054fdac7075b8b94d99f67e2e1b122e2685549?

**Comment by quinnj** [comment]
Yes, should be fixed on master; things are a bit in flux right now as I'm doing a swath of updates across CSV.jl, SQLite.jl, DataStreams.jl, ODBC.jl, and Feather.jl. The "Project DECOUPLE" is finally happening, where the individual "DataStreams" packages won't actually have to depend on each other anymore and there will be a true actual interface for streaming data. I've got it mostly working (i.e. tests passing) on all the master branches, but I still want to take it to the next round of polish with more robust tests, benchmarks and slick new docs (as I've finally figured out with CSV.jl). More fun to come in the next few weeks.
